mypy/main.py

```diff
@@ -513,6 +523,32 @@ def add_invertible_flag(flag: str,
     if options.quick_and_dirty:
         options.incremental = True
 
+    # try setting a valid Python executable based on a specified version
+    if options.python_version:
+        if not options.python_executable:
+            try:
+                output = subprocess.check_output(
+                    [python_executable_prefix.format(options.python_version)],
+                    stderr=subprocess.STDOUT).decode()
+            except FileNotFoundError:
+                output = ''
+            if output.startswith('Python '):
+                options.python_executable = python_executable_prefix.format(options.python_version)
+
+    # Set Python version if given Python executable, but no version
+    if options.python_executable:
+        if not options.python_version:
+            check = subprocess.check_output([options.python_executable, '-V'],
+                                            stderr=subprocess.STDOUT).decode()
+            if not check.startswith('Python '):
+                print("Mypy could not use the Python executable: {}".format(
+                    options.python_executable), file=sys.stderr)
+                sys.exit(2)
+            ver = re.fullmatch(r'Python (\d)\.(\d)\.\d\s*', check)
```

Comment:
You maybe want `\d+` here to match things like `2.7.14`, or `3.10.x` if that comes after `3.9.x`

Reply:
Good point. Thinking of 3.10.x, never thought I'd see the day!